tests/gnrc_netif_ipv6_wait_for_global_address/main.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Tests notification for global address
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <stdio.h>
+
+#include "embUnit.h"
+#include "embUnit/embUnit.h"
+#include "net/ipv6.h"
+#include "net/gnrc/netif.h"
+#include "net/gnrc/netif/raw.h"
+#include "net/netdev_test.h"
+#include "xtimer.h"
+
+#define TEST_NETIF_NUMOF    2
+#define TEST_NETIF_PRIO     3
+
+static netdev_test_t netdev_test[TEST_NETIF_NUMOF];
+static gnrc_netif_t netif_test[TEST_NETIF_NUMOF];
+static char netif_stack[TEST_NETIF_NUMOF][THREAD_STACKSIZE_DEFAULT];
+static char adder_stack[THREAD_STACKSIZE_DEFAULT];
+static mutex_t adder_mtx;
+
+static void *_adder_thread(void *netif)
+{
+    const ipv6_addr_t addr =  {{ 0x20, 0x01, 0x0d, 0xd8,
+                                 0x00, 0x00, 0x00, 0x00,
+                                 0x00, 0x00, 0x00, 0x00,
+                                 0x00, 0x00, 0x00, 0x01 }};
+
+    xtimer_msleep(10);
+    gnrc_netif_ipv6_addr_add(netif, &addr, 64, GNRC_NETIF_IPV6_ADDRS_FLAGS_STATE_VALID);
+    xtimer_msleep(5);
+    gnrc_netif_ipv6_addr_remove(netif, &addr);
```

Comment:
It would be cleaner, if this removial would be done in a `tear_down` function, and each test case being, well, their own test case (i.e. in a separate function).